src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportBasicService.java

```diff
@@ -104,40 +104,40 @@ public ProgrammingExercise importProgrammingExerciseBasis(final ProgrammingExerc
 
         // Hints, tasks, test cases and static code analysis categories
         Map<Long, Long> newHintIdByOldId = exerciseHintService.copyExerciseHints(templateExercise, newExercise);
-        programmingExerciseRepository.save(newExercise);
-        Map<Long, Long> newTestCaseIdByOldId = importTestCases(templateExercise, newExercise);
-        Map<Long, Long> newTaskIdByOldId = importTasks(templateExercise, newExercise, newTestCaseIdByOldId);
-        updateTaskExerciseHintReferences(templateExercise, newExercise, newTaskIdByOldId, newHintIdByOldId);
-        importSolutionEntries(templateExercise, newExercise, newTestCaseIdByOldId, newHintIdByOldId);
+
+        final ProgrammingExercise importedExercise = programmingExerciseRepository.save(newExercise);
```

Comment:
If you add a final here, please also add it to the other variable declarations in the method.